Jenkinsfile

```diff
@@ -0,0 +1,296 @@
+
+        sh "${xbuild} Platform.XamarinIOS/Tests.XamarinIOS/Tests.XamarinIOS.csproj /p:Configuration=${configuration} /p:Platform=iPhoneSimulator /p:SolutionDir=\"${workspace}/\""
+
+        stash includes: "Platform.XamarinIOS/Realm.Sync.XamarinIOS/bin/iPhoneSimulator/${configuration}/Realm.Sync.*", name: 'nuget-ios-sync'
+
+        dir("Platform.XamarinIOS/Tests.XamarinIOS/bin/iPhoneSimulator/${configuration}") {
+          stash includes: 'Tests.XamarinIOS.app/**/*', name: 'ios-tests-sync'
+        }
+      }
+    },
+    'Android': {
+      node('xamarin-mac') {
+        getArchive()
+
+        dir('wrappers') {
+          withEnv(["NDK_ROOT=${env.HOME}/Library/Developer/Xamarin/android-ndk/android-ndk-r10e"]) {
+            sh "make android${wrapperConfigurations[configuration]}"
+          }
+        }
+
+        stash includes: "wrappers/build/${configuration}-android/**/*", name: 'android-wrappers-sync'
+      }
+      node('xamarin-mac') {
+        getArchive()
+        def workspace = pwd()
+
+        unstash 'android-wrappers-sync'
+
+        sh "${nuget} restore Realm.sln"
+
+        dir('Platform.XamarinAndroid/Tests.XamarinAndroid') {
+          sh "${xbuild} Tests.XamarinAndroid.csproj /p:Configuration=${configuration} /t:SignAndroidPackage /p:AndroidUseSharedRuntime=false /p:EmbedAssembliesIntoApk=True /p:SolutionDir=\"${workspace}/\""
+          dir("bin/${configuration}") {
+            stash includes: 'io.realm.xamarintests-Signed.apk', name: 'android-tests-sync'
+          }
+        }
+
+        stash includes: "Platform.XamarinAndroid/Realm.Sync.XamarinAndroid/bin/${configuration}/Realm.Sync.*", name: 'nuget-android-sync'
+      }
+    },
+    'PCL': {
+      node('xamarin-mac') {
+        getArchive()
+        sh "${nuget} restore Realm.sln"
+        sh "${xbuild} Platform.PCL/Realm.Sync.PCL/Realm.Sync.PCL.csproj /p:Configuration=${configuration}"
+        stash includes: "Platform.PCL/Realm.Sync.PCL/bin/${configuration}/Realm.Sync.*", name: 'nuget-pcl-sync'
+      }
+    }
+  )
+}
+
+stage('Test without sync') {
+  parallel(
+    'iOS': {
+      node('osx') {
+        unstash 'ios-tests-nosync'
+
+        dir('Tests.XamarinIOS.app') {
+          sh 'mkdir -p fakehome/Documents'
+          sh "HOME=`pwd`/fakehome DYLD_ROOT_PATH=`xcrun -show-sdk-path -sdk iphonesimulator` ./Tests.XamarinIOS --headless"
+          publishTests 'fakehome/Documents/TestResults.iOS.xml'
+        }
+      }
+    },
+    'Android': {
+      node('android-hub') {
+        sh 'rm -rf *'
+        unstash 'android-tests-nosync'
+        sh 'adb devices'
+        sh 'adb devices | grep -v List | grep -v ^$ | awk \'{print $1}\' | parallel \'adb -s {} uninstall io.realm.xamarintests; adb -s {} install io.realm.xamarintests-Signed.apk; adb -s {} shell am instrument -w -r io.realm.xamarintests/.TestRunner; adb -s {} shell run-as io.realm.xamarintests cat /data/data/io.realm.xamarintests/files/TestResults.Android.xml > TestResults.Android_{}.xml\''
```

Comment:
This is a long line. Sure part of it would not be better of in a shell script? anyway, you could write it in multiple lines.